projects/plugins/boost/app/assets/src/js/features/image-cdn/quality-settings/quality-settings.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { createInterpolateElement, useCallback } from '@wordpress/element';
+import CollapsibleMeta from '../collapsible-meta/collapsible-meta';
+import { navigate } from '$lib/utils/navigate';
+import { __, sprintf } from '@wordpress/i18n';
+
+import styles from './quality-settings.module.scss';
+import { IconTooltip } from '@automattic/jetpack-components';
+import QualityControl from '../quality-control/quality-control';
+import { DataSyncProvider, useDataSync } from '@automattic/jetpack-react-data-sync-client';
+import { type QualityConfig, imageCdnSettingsSchema } from '../lib/stores';
+import { z } from 'zod';
+
+type QualitySettingsProps = {
+	isPremium: boolean;
+};
+
+const QualitySettings = ( { isPremium }: QualitySettingsProps ) => {
+	if ( ! isPremium ) {
+		return createInterpolateElement(
+			__( `For more control over image quality, <link>upgrade now!</link>`, 'jetpack-boost' ),
+			{
+				link: (
+					// eslint-disable-next-line jsx-a11y/anchor-has-content, jsx-a11y/anchor-is-valid
+					<a
+						href="#"
+						onClick={ () => {
+							navigate( 'upgrade' );
+						} }
+					/>
```

Comment:
This upgrade link doesn't work. When you click it, it just loads `href="#"`. To make it work, you'd need to take an `event` parameter to your click handler and call `event.preventDefault()`.

Instead, why not... 
```suggestion
					<a href="#upgrade" />
```

If we do have to use a click handler / call navigate gracefully, then we might want a helper function in `$lib/utils/navigate` which returns an anchor tag ready to go with the right props for use in createInterpolateElement to navigate to another page.

Reply:
Lovely - NavigationLink component does the job nicely, fixes the bug and is reusable. 🥇